pegasus/helpers/section_api_helpers.rb

```diff
@@ -252,6 +252,14 @@ def self.valid_courses(user_id = nil)
         end
   end
 
+  def self.progress_disabled_courses(user_id = nil)
+    disabled_courses = valid_courses(user_id).select do |course|
+      script_name = Script.find(course[:id]).name
+      !Gatekeeper.allows('postMilestone', where: {script_name: script_name}, default: true)
+    end
```

Comment:
AFAIK, querying the `Gatekeeper` is cheap. Perhaps @wjordan can say for certain.